core/sqf/inc/cextdecs/cextdecs.h

```diff
@@ -289,6 +289,10 @@ _declspec(dllimport) short   DNUMOUT
 
 _declspec(dllimport) short   ENDTRANSACTION();
 
+//errStr is allocated if errlen is not zero. Caller must deallocate errStr.
+//Rest of functionality same as ENDTRANSACTION();
+_declspec(dllimport) short   ENDTRANSACTION_ERR(char *&errStr, int &errlen);
```

Comment:
This would mean that the errStr is allocated by the TM functions using its own heap management scheme or the system memory. The caller wouldn't be aware of how to deallocate if it is anything other than system memory